code/src/java/pcgen/gui2/facade/Gui2InfoFactory.java

```diff
@@ -859,15 +859,15 @@ private HtmlInfoBuilder getEquipmentHtmlInfo(Equipment equip)
 
 		Integer a = EqToken.getMaxDexTokenInt(pc, equip);
 
-		if (a.intValue() != Constants.MAX_MAXDEX)
```

Comment:
So the question on these (where it's == or !=) is do compilers unconditionally error on this situation of Integer == Integer?  They haven't in the past, and this is where automated boxing assumptions get dangerous... if the Constant is changed, this code is vulnerable.  Therefore removing the .intValue() increases the fragility of this code.  I think this is another point for discussion.